documentation/new-version-guideline.md

```diff
@@ -0,0 +1,342 @@
+	// advance item
+	iter.index++
+	return item, nil
+}
+
+func NewIterator[TItem any, TPage PageConstraint[TItem]](pager *runtime.Pager[TPage]) *Iterator[TItem, TPage] {
+	return &Iterator[TItem, TPage]{
+		pager: pager,
+	}
+}
+```
+
+***Usage***
+```go
+ctx := context.TODO() // your context
+iter := NewIterator[armresources.ResourceGroup](rgClient.NewListPager(nil))
+for iter.More() {
+    rg, err := iter.NextItem(ctx)
+    if err != nil {
+        // handle error...
+    }
+    // dealing with `rg`
+}
+```
+
+### Reference
+
+For more information, you can refer to [design guidelines of Paging](https://azure.github.io/azure-sdk/golang_introduction.html#methods-returning-collections-paging) and [API reference of pager](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime#Pager).
+
+## Long-Running Operations
+
+### General usage
+
+Some operations can take a long time to complete. Azure introduces the long-running operations (LROs) to do such operations asynchronously. You need to use Begin* to start an LRO. It will return a poller that can used to keep polling for the result until LRO is done. For examples:
+
+```go
+ctx := context.TODO() // your context
+poller, err := client.BeginCreate(ctx, "resource_identifier", "additonal_parameter", nil)
+if err != nil {
+    // handle error...
+}
+resp, err = poller.PollUntilDone(ctx, 5 * time.Second)
+if err != nil {
+    // handle error...
+}
+// dealing with `resp`
+```
+
+> NOTE: You will need to pass a polling interval to `PollUntilDone` and tell the poller how often it should try to get the status. This number is usually small but it's best to consult the [Azure service documentation](https://docs.microsoft.com/azure/?product=featured) on best practices and recommended intervals for your specific use cases.
+
+### Resume Tokens
+
+Pollers provide the ability to serialize their state into a "resume token" which can be used by another process to recreate the poller. For example:
+
+```go
+import "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
+```
+
+```go
+ctx := context.TODO() // your context
+poller, err := client.BeginCreate(ctx, "resource_identifier", "additonal_parameter", nil)
+if err != nil {
+    // handle error...
+}
+token, err := poller.ResumeToken()
+if err != nil {
+    // handle error...
+}
+
+// ... 
+
+// recreate the poller from the token
+poller, err = client.BeginCreate(ctx, "", "", &armresources.ResourceGroupsClientBeginCreateOptions{
+    ResumeToken: token,
+})
+resp, err = poller.PollUntilDone(ctx, 5 * time.Second)
+if err != nil {
+    // handle error...
+}
+// dealing with `resp`
+```
+
+> NOTE: A token can only be obtained for a poller that's not in `Succeeded`, `Failed` or `Canceled` state. Each time you call `poller.Poll()`, the token might change because of the LRO state's change. So if you need to cache the token for crash consistency, you need to update the cache when calling `poller.Poll()`.
+
+### Synchronized wrapper
```

Comment:
I don't see any mention that this wrapper makes the returned error ambiguous.

Reply:
Added a note about this.